### PR TITLE
Helical Scan first version

### PR DESCRIPTION
**Pull request by bolmsten**
This PR is for creating lines and making Helical scans. It also has some minor GUI fixes to the todo queue

### CODE REVIEW

File: mxcube3/routes/qutils.py [hunk 1]
```diff
@@ -359,6 +359,23 @@ def set_dc_params(model, entry, task_data):
                 _cpos.index = int(params['point'])
                 acq.acquisition_parameters.centred_position = _cpos
 
+    if params["helical"]:
```
Comment (marcus-oscarsson) [inline mxcube3/routes/qutils.py]: Hm, this was is already added by Mikels PR #283 are you sure that there is not a problem with the PRs ?

Comment (bolmsten) [inline mxcube3/routes/qutils.py]: Good point, mine is before Mikels and maybe should be reviewed first

**Comment by marcus-oscarsson** [comment]
Hey

I noticed some linting errors. I dont really mind its jut max-len you can perhaps fix them if you have time. Otherwise we will fix them later ...

Cheers,
Marcus

**Comment by bolmsten** [comment]
I will try to fix them later on as I am planning a refactoring of the shape class

**Comment by marcus-oscarsson** [comment]
Thanks Fredrik !,

The code is general clean and easy to read.

The modifier you are using right now seems to be Shift I think its maybe more common to use CTRL
for selecting multiple items ?

I also remembered that we would avoid to have the draw line step explicitly done by the user. But that
a line is drawn when two points are selected and helical chosen as method. We might have changed our minds and I'm simply not aware of it ? The reason was simply to save the user from drawing the line.

These are just reflections I think the code is fine in general, ill give Matias a chance to take a look before we merge.

**Comment by bolmsten** [comment]
Thank you @marcus-oscarsson ,

I honestly do not remember the conversation regarding the line. I have briefly discussed with Mikel now and we feel like doing it like that is mixing styles if compared to creating a point. 

To add datacollection or charaterisation one first creates a point then adds tasks, our idea would be to do the same with the helical. This is in order to keep the styles the same but we can discuss best solutions.

Regarding the ctrl vs shift: I agree with you but the shift is built in the library of fabric js so I did no changes at all. We could look into how to change this.

**Comment by marcus-oscarsson** [comment]
Okey,

I'm not so sure that it would be mixing styles, the actions in the drop down for a set of selected positions should be Add Collection, Add Characterization, Add Helical and Remove. So it makes sense that you can add a helical directly since you already can add collections and characterizations. The line is currently only used by the helical and explicitly adding it makes little sense. That's at least how we thought, but maybe you have something else in mind.

**Comment by meguiraun** [comment]
Hej, I thought that we _operate_ on geometric elements, some actions are related to points and others to lines. In any case, could we accept this PR as it is now and let our product owners decide about right click on points/lines/whatever? We can modify this behavior later on, it should only affect helical.

**Comment by mguijarr** [comment]
At least one commit in this branch is also present in Mikel's branch for characterisation : see [this one](https://github.com/meguiraun/mxcube3/commit/5bbd22a1e51f09bbfd99a97542cfe60be3e82209).
Could you please rebase your branch on master to have a cleaner PR ? It is difficult to follow.
Thanks in advance !

**Comment by mguijarr** [comment]
@bolmsten, @meguiraun : the behaviour Marcus described above comes from our proposal last time you were at ESRF, see page 8 of the pdf document. If you don't have it I can send it to you again.

We talked about it at that time. This design is validated by Daniele of course. Do Marjolein have any strong opinion against it ?

As you propose Mikel we can accept the PR as it is regarding the 'right click' behavior, but it has to be fixed afterwards.

**Comment by meguiraun** [comment]
no no no... It was just my opinion. I looked at the doc, my head can not contain much info... We can adapt to it later on.

**Comment by mguijarr** [comment]
Thanks for removing the server io thing. But you added a commit to remove it, whereas it should not have been there at first... In addition you didn't rebase so we still have those ugly 'merge commits'.

**Comment by bolmsten** [comment]
Rebased and implemented the way of adding helical as suggested

File: mxcube3/routes/qutils.py [hunk 2]
```diff
@@ -23,7 +24,6 @@ def __reduce__(self):
 def node_index(node):
     """
     Get the position (index) in the queue, sample and node id of node <node>.
-
```
Comment (marcus-oscarsson) [inline mxcube3/routes/qutils.py]: I would prefer if we keep the new lines, it makes the doc strings more readable

**Comment by mguijarr** [comment]
Thanks a lot !

**Pull request merged**